### ISSUE TITLE
ModuleNotFoundError: No module named 'dotenv'

### ISSUE THREAD

**Issue by mangeshsable1975** [kalpita01/nsepython#1]
ModuleNotFoundError: No module named 'dotenv'

**Comment by kalpita01** [comment]
try installing dotenv module again in your machine, you can do that by typing this command in your terminal
python -m pip install python-dotenv

**Comment by mangeshsable1975** [comment]
i install donnov module after pip installing pynse,
then also having error ``*Defaulting to user installation because normal
site-packages is not writeable*





*Collecting https://github.com/mangeshsable1975/pynse.git
<https://github.com/mangeshsable1975/pynse.git>  Downloading
https://github.com/mangeshsable1975/pynse.git
<https://github.com/mangeshsable1975/pynse.git>  ERROR: Cannot unpack file
C:\Users\admin\AppData\Local\Temp\pip-unpack-894y89_q\pynse.git (downloaded
from C:\Users\admin\AppData\Local\Temp\pip-req-build-z39msn03,
content-type: text/html; charset=utf-8); cannot detect archive formatERROR:
Cannot determine archive format of
C:\Users\admin\AppData\Local\Temp\pip-req-build-z39msn03WARNING: You are
using pip version 20.2.3; however, version 21.2.4 is available.You should
consider upgrading via the 'c:\program files\python38\python.exe -m pip
install --upgrade pip' command.*

**Comment by kalpita01** [comment]
You're facing this error as you don't have git installed in your machine. 
You can download it over here - https://git-scm.com/downloads

I would suggest you clone the repository once again in your machine, after that install pynse and pyTelegramBotAPI modules. You can do that by typing these commands in your terminal.
pip install --upgrade git+https://github.com/StreamAlpha/pynse.git
pip install pyTelegramBotAPI

Note:- 
- Make sure you have git installed on your machine before typing the above commands. Otherwise, you are gonna face the same errors that you're facing now.
-  Also make sure to enter your bot's API within the code as well.

**Comment by mangeshsable1975** [comment]
sorry kalpita
my computer was not working.
sorry for Delay
i have follow the steps as you suggest in earlier mail

pip install --upgrade git+https://github.com/StreamAlpha/pynse.git
pip install pyTelegramBotAPI
both steps installed without errors
& i had  enter bot's API within the code
stell have some error

C:\Users\admin\AppData\Local\Programs\Python\Python39\python.exe: Error
while finding module specification for 'pyramid.scripts.pserve'
(ModuleNotFoundError: No module named 'pyramid')

please suggest solution


thanking you

mangesh

**Comment by mangeshsable1975** [comment]
ERROR IN DEBUGPY\LAUNCHER
[image: image.png]

**Comment by mangeshsable1975** [comment]
good evening kalpita
nsepython bot is working now
thanks,
i would like to suggest some modification,
  * can we used this program on cloud based.
  * can we kept active and have a sleep mode as required as (5min, 10min.
interval)
  * This is tracking data which is dynamic, so can we add future trades,
filters as we can use for any strick price (ex. nifty 18000 pe = 100 if
true, a notification/ massage send to bot

thanking you

**Comment by kalpita01** [comment]
Yes, indeed these features can be implemented. Thanks for the suggestions, will try to work on these suggestions in the future. Feel free to make a pull request if you can add any new features.
